src/main/java/uk/gov/hmcts/reform/prl/services/citizen/CaseService.java

```diff
@@ -378,4 +397,200 @@ public CaseDetails withdrawCase(CaseData caseData, String caseId, String authTok
+            eventRequestData,
+            caseDataContent,
+            caseId,
+            false
+        );
+        return ResponseEntity.status(HttpStatus.OK).body("party flags updated");
+    }
+
+    private List<Element<FlagDetail>> convertFlags(List<Element<FlagDetailRequest>> details) {
+        List<Element<FlagDetail>> flagDetails = new ArrayList<>();
+
+        for (Element<FlagDetailRequest> detail : details) {
+            FlagDetail flagDetail = FlagDetail.builder().name(detail.getValue().getName())
+                .name_cy(detail.getValue().getName_cy())
+                .subTypeValue(detail.getValue().getSubTypeValue())
+                .subTypeValue_cy(detail.getValue().getSubTypeValue_cy())
+                .subTypeKey(detail.getValue().getSubTypeKey())
+                .otherDescription(detail.getValue().getOtherDescription())
+                .otherDescription_cy(detail.getValue().getOtherDescription_cy())
+                .flagComment(detail.getValue().getFlagComment())
+                .flagComment_cy(detail.getValue().getFlagComment_cy())
+                .flagUpdateComment(detail.getValue().getFlagUpdateComment())
+                .dateTimeCreated(detail.getValue().getDateTimeCreated())
+                .dateTimeModified(detail.getValue().getDateTimeModified())
+                .path(detail.getValue().getPath())
+                .hearingRelevant(detail.getValue().getHearingRelevant())
+                .flagCode(detail.getValue().getFlagCode())
+                .status(detail.getValue().getStatus())
+                .availableExternally(detail.getValue().getAvailableExternally())
+                .build();
+
+            if (null != detail.getId()) {
+                flagDetails.add(element(detail.getId(), flagDetail));
+            } else {
+                flagDetails.add(element(flagDetail));
+            }
+        }
+
+        return flagDetails;
+    }
+
+    public Optional<String> getPartyExternalCaseFlagField(String caseType, PartyEnum partyType, Integer partyIndex) {
+
+        Optional<String> partyExternalCaseFlagField = Optional.empty();
+        boolean isC100Case = C100_CASE_TYPE.equalsIgnoreCase(caseType);
+
+        if (PartyEnum.applicant == partyType) {
+            partyExternalCaseFlagField
+                = Optional.ofNullable(partyLevelCaseFlagsService.getPartyCaseDataExternalField(
+                caseType,
+                isC100Case ? PartyRole.Representing.CAAPPLICANT : PartyRole.Representing.DAAPPLICANT,
+                partyIndex
+            ));
+        } else if (PartyEnum.respondent == partyType) {
+            partyExternalCaseFlagField
+                = Optional.ofNullable(partyLevelCaseFlagsService.getPartyCaseDataExternalField(
+                caseType,
+                isC100Case ? PartyRole.Representing.CARESPONDENT : PartyRole.Representing.DARESPONDENT,
+                partyIndex
+            ));
+        }
+
+        return partyExternalCaseFlagField;
+    }
```

Comment:
Can we move this to PartyLevelCaseFlagService please. planning to reuse this code for amend and return journey.